docs/querying/datasource.md

```diff
@@ -211,6 +219,50 @@ Inline datasources are not available in Druid SQL.
 Refer to the [Query execution](query-execution.md#inline) page for more details on how queries are executed when you
 use inline datasources.
 
+### Dynamic table append `TABLE(APPEND(...))`
```

Comment:
Consider also including a line / link to this content from the [SQL syntax doc](https://druid.apache.org/docs/latest/querying/sql#union-all) in https://github.com/apache/druid/blob/master/docs/querying/sql.md?plain=1#L322

Reply:
referenced updated the query and referenced [table-level union](sql.md#table-level) as that's more closer to what the user might be working with
thank you!

Reply:
I've added a backreferencing paragraph to that as well